common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -142,12 +143,53 @@ public static Geometry transform(Geometry geometry, String sourceCRS, String tar
 
     public static Geometry transform(Geometry geometry, String sourceCRS, String targetCRS, boolean lenient)
         throws FactoryException, TransformException {
-        CoordinateReferenceSystem sourceCRScode = CRS.decode(sourceCRS);
-        CoordinateReferenceSystem targetCRScode = CRS.decode(targetCRS);
+
+        CoordinateReferenceSystem sourceCRScode = parseCRSString(sourceCRS);
+        CoordinateReferenceSystem targetCRScode = parseCRSString(targetCRS);
         MathTransform transform = CRS.findMathTransform(sourceCRScode, targetCRScode, lenient);
         return JTS.transform(geometry, transform);
     }
 
+    private static CoordinateReferenceSystem parseCRSString(String CRSString) throws FactoryException{
+        if (checkCRSCodeFormat(CRSString) ){
+            return CRS.decode(CRSString);
+        }
+        else if (checkCRSWKTFormat(CRSString)){
+            return CRS.parseWKT(CRSString);
+        }
```

Comment:
Since these check methods just try to do the decoding, should this just be something like
```suggestion
        try {
            return CRS.decode(CRSString);
        } catch (FactoryException ex) {
            try {
                return CRS.parseWKT(CRSString);
            }  catch (FactoryException ex2) {
                // ...raise some exception maybe from one or the other or some custom thing explaining can't be parsed
        }
```